functions/Copy-SqlMaintenancePlan.ps1

```diff
@@ -0,0 +1,237 @@
+		[switch]$Force
+	)	
+	DynamicParam { if ($source) { return (Get-ParamSqlMaintenancePlans -SqlServer $source -SqlCredential $SourceSqlCredential) } }
+	
+	BEGIN
+	{
+		$plans = $psboundparameters.MaintenancePlans
+		$exclude = $psboundparameters.Exclude
+
+		$sourceserver = Connect-SqlServer -SqlServer $Source -SqlCredential $SourceSqlCredential
+		$destserver = Connect-SqlServer -SqlServer $Destination -SqlCredential $DestinationSqlCredential
+
+		$source = $sourceserver.DomainInstanceName
+		$destination = $destserver.DomainInstanceName
+	}
+	
+	PROCESS
+	{
+		$qryPlanNames = "SELECT plans.[name] AS plan_name, plans.id AS plan_id
+			FROM msdb.dbo.sysmaintplan_plans AS plans"
+
+		$sourceMaintPlanList = $sourceserver.ConnectionContext.ExecuteWithResults($qryPlanNames).Tables.Rows
+		$destMaintPlanList = $destserver.ConnectionContext.ExecuteWithResults($qryPlanNames).Tables.Rows
+
+		foreach ($plan in $sourceMaintPlanList)
+		{
+			$planname = $plan.plan_name
+			$planid = $plan.plan_id
+
+			if ($plans.count -gt 0 -and $plans -notcontains $planname -or $exclude -contains $planname)
+			{ continue }
+
+			if ($destMaintPlanList.plan_name -contains $plans)
+			{
+				if ($force -eq $false)
+				{
+					Write-Warning "[Plan: $planname] exist at destination. Use -Force to drop and migrate."
+					continue
+				}
+				else
+				{
+					if ($Pscmdlet.ShouldProcess($destination, "Dropping Maintenance Plan: $planname on target $destination"))
+					{
+					#folderid (08AA12D5-8F98-4DAB-A4FC-980B150A5DC8) = Maintenance Plans folder under MSDB
+						#
+						$qryDropMaintenancePlan = "
+						EXECUTE msdb.dbo.sp_maintplan_delete_plan 
+							@plan_id=N'{$planid}' 
+						EXECUTE [msdb].[dbo].[sp_ssis_deletepackage] 
+							@name = N'$planname', 
+							@folderid = '08AA12D5-8F98-4DAB-A4FC-980B150A5DC8'
+						"
+						try
+						{
+							Write-Output "Dropping Maintenance Plan $planname on $destination"
+							$destserver.ConnectionContext.ExecuteNonQuery($qryDropMaintenancePlan) > $null
+						}
+						catch
+						{
+							Write-Exception $_
+							continue
+						}
+					}
+				}
+			}
+			if ($Pscmdlet.ShouldProcess($destination, "Creating Maintenance Plan: $planname"))
+			{
+				try
+				{
+					Write-Output "Copying Maintenance Plan $planname"
+					$qryMaintMetaData = "
+					SELECT [name], [id], [description], [createdate], [folderid], [ownersid],
+					CAST(
+					   CAST(
+						  REPLACE(
+							 CAST(
+								CAST(packagedata AS varbinary(max)
+								) AS varchar(max)
+							 ),
+							 '$source','$destination')
+					   AS xml)
+					AS varbinary(max) ) AS packagedata,
+					packageformat, packagetype, vermajor, verminor, verbuild, vercomments, verid, isencrypted, readrolesid, writerolesid
+					FROM [msdb].[dbo].[sysssispackages]
+					WHERE name = '$planname'"
+
+					Write-Verbose "[SELECT] query - Maintenance metadata"
+					Write-Verbose $qryMaintMetaData
```

Comment:
All $sql should be output in Write-Debug. We found that writing it in verbose was overwhelming.